### ISSUE TITLE
MinGW-64 conflict with sim_disk.c:CreateVirtualDisk

### ISSUE THREAD

**Issue by bscottm** [open-simh/simh#52]
Respectfully request that open-simh consider renaming. This has been a problem for going on 3 years: `CreateVirtualDisk` is a name conflict when compiling with MinGW64 GCC, since the MinGW64 GCC 8 series. 

@markpizz will gladly tell me to go tell the MinGW maintainers to Do The Right Thing and not include `virtdisk.h` from within whichever header file the include is currently embedded. Asking MinGW to make a change is not feasible, given the size of the user base and likely other code depending on the current MinGW header structure.

**Comment by markpizz** [comment]
The original purpose of building simh with MinGW was solely due to the fact that when this project got started, MinGW was the only free tool chain which was available on Windows.  Microsoft came out with freely available C compiler with Visual Studio 2005.  

Several years after the free Visual Studio 2008 was available, MinGW32 builds have been on a "if it works for you, great", but not an absolute requirement.  Over the last 10 years, I got a handful of minor inputs to the makefile from ONE person who was OK with the "if it works for you" mode.

As for MinGW64, it has never been supported since, as @bscottm points out, there is an incorrect nesting include files in the MinGW64 codebase.  The point of MinGW (32 or 64) is/was to expose the Win32 API set via gcc so folks needing a free compiler could directly build native windows applications.

As for the benefits of MinGW64 over MinGW32, there are no simh simulators which produce measurably different performance when compiled with an x64 compiler vs the x86 32bit one.

Migrating to the cmake build system needn't build with extended compilers over the makefile approach.

A somewhat related issue is attempting to build under cygwin.  Cygwin is a Posix layer over the Win32 APIs and as such only Posix APIs are available.  Builds may work (under the "if it works for you" mentioned above), but, in general functionality will be less or less precise than a native built simulators.

As for: 
> @markpizz will gladly tell me to go tell the MinGW maintainers to Do The Right Thing and not include virtdisk.h from within whichever header file the include is currently embedded. Asking MinGW to make a change is not feasible, given the size of the user base and likely other code depending on the current MinGW header structure.

Since the point of the MinGW approach is to precisely mimic the native Windows API functionality with a free compiler, changing the include nesting would only correct what is broken.  I suggested this to @bscottm back in https://github.com/simh/simh/issues/785

Since he's back with precisely the same suggestion 2.5 years later without any justification from those who could fix this the right way, the reasoning to not change anything should still be the same.

**Comment by bscottm** [comment]
@markpizz: If `open-simh` should _never_ build using MinGW64, then let's make that explicit and say that the CMake MinGW generators are unsupported as the result of this issue.

Asking the MinGW maintainers to fix their headers is futile. I don't need justification -- `open-simh` and `simh` are small projects and David isn't going to kill Goliath in this case. I'm taking an entirely practical stance.

**Comment by markpizz** [comment]
Like I pointed out back in https://github.com/simh/simh/issues/785, I fixed problems with upstream dependencies, reported them and, in fact, the changes I reported upstream were indeed included in subsequent versions. 

Had you reported the MinGW64 issue 2,5 years ago, the problem would have been fixed already and we wouldn't be having this discussion.  Upstream wouldn't be fixing the problem to help simh, but to make their project more correct since they're providing Win32 APIs and what's been done just didn't happen to be noticed by others.  A bug is a bug not a David vs Goliath problem.  Surely it is a small bug, so it certainly wouldn't specifically motivate a release, but like the cases I reported, the fix would eventually be included in the next release.

> @markpizz: If open-simh should never build using MinGW64, then let's make that explicit and say that the CMake MinGW generators are unsupported as the result of this issue. windows.h on both MinGW32 and MinGW64 include virtdisk.h, so both environments will have to be explicitly unsupported.

The demand for MinGW(32 or 64) has been very close to 0 for the last 12 years.  You're the one creating the only demand for MinGW64, so making one or both unsupported would hardly seem to bother anyone.

Folks should read through https://github.com/simh/simh/issues/785 and look at @rcornwell and my comments about the value vs the complexity of adding more tool-chains to the support matrix.  Given the free compilers from Microsoft since 2005, the value for other compilers really isn't there.
